packages/core/src/hooks/useSwipe.tsx

```diff
@@ -0,0 +1,58 @@
+/*
+ * Axelor Business Solutions
+ *
+ * Copyright (C) 2023 Axelor (<http://axelor.com>).
+ *
+ * This program is free software: you can redistribute it and/or  modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {Dimensions} from 'react-native';
+
+const windowWidth = Dimensions.get('window').width;
+
+export function useSwipe(
```

Comment:
please export this hook from package index to use it outside of this package, 
Also I think it could be better to move it to UI package next to AnimationUtils as animations are mostly UI